### ISSUE TITLE
Warn when data source `url_params` already exist

### ISSUE THREAD

**Issue by bcamper** [tangrams/tangram#515]
Sometimes a data source specifies `url_params` that already exist on the query string for the source's `url`. Without getting into full-on query string parsing/splicing (and possibly introduce new bugs), we may want to trigger a naive warning when it appears that one of the params included in `url_params` is already present on the source's `url` query string.

**Comment by pnorman** [comment]
How would you detect this without parsing the query string? Is a regex enough?

**Comment by bcamper** [comment]
Indeed that's why I'm hesitant to do too much here. But perhaps a regex matching the query string for each of the keys in `url_params`.

**Comment by bcamper** [comment]
The specific example where this arose is with duplicate/conflicting API keys for Mapzen tiles, but could happen more generally for similar cases.

**Comment by pnorman** [comment]
> The specific example where this arose is with duplicate/conflicting API keys for Mapzen tiles, but could happen more generally for similar cases

I figured this is probably what it was. `[?&]$KEY=` might work as a regex (ignoring escaping probably needed). Or do you need `?$KEY=` OR `?.*&$KEY=`? Given we're within a browser, is there not some function that does URL parsing?

**Comment by bcamper** [comment]
From looking at https://davidwalsh.name/query-string-javascript, there is a newish builtin, [URLSearchParams](http://caniuse.com/#search=URLSearchParams), but it is Chrome/FF only right now. But there is a small function offered for basic single parameter checking:

```
function getUrlParameter(name) {
    name = name.replace(/[\[]/, '\\[').replace(/[\]]/, '\\]');
    var regex = new RegExp('[\\?&]' + name + '=([^&#]*)');
    var results = regex.exec(location.search);
    return results === null ? '' : decodeURIComponent(results[1].replace(/\+/g, ' '));
};
```

**Comment by pnorman** [comment]
Yep, url parsing is not trivial. But it's good to see that someone has written the code.

Alternately, since it's a warning, URLSearchParams might be an option - the failure mode is not outputting a warning to the developer via the console, not failing to rendering. Safari 10.1 will support it too.

**Comment by bcamper** [comment]
Added in eae56dcf308083395c4f30d091a3a31cf2280c02

**Comment by bcamper** [comment]
Released in [v0.12.0](https://github.com/tangrams/tangram/releases/tag/v0.12.0)
